### PR TITLE
:hammer: Add `SecurityContextConstraints` for OpenShift

### PR DESCRIPTION
**Pull request by kcns008**
KSOS-helm

### CODE REVIEW

File: helm-chart/values.yaml [hunk 1]
```diff
@@ -6,12 +6,14 @@ tap:
...
+      srvPort: 30000
     hub:
-      port: 8898
-      srvPort: 8898
+      port: 30001
+      srvPort: 30001
     front:
-      port: 8899
+      port: 30002
+      srvPort: 30001
```
Comment (mertyildiran) [inline helm-chart/values.yaml:16]: `front.srvPort` is not templated.

File: helm-chart/values.yaml [hunk 2]
```diff
@@ -6,12 +6,14 @@ tap:
...
+      port: 30000
+      srvPort: 30000
     hub:
-      port: 8898
-      srvPort: 8898
+      port: 30001
+      srvPort: 30001
     front:
-      port: 8899
+      port: 30002
```
Comment (mertyildiran) [inline helm-chart/values.yaml:15]: Don't change the defaults.

**Review by mertyildiran — CHANGES_REQUESTED** [review]
The addition of `hostPort` is missing.

File: helm-chart/values.yaml (hunk 2 above)
Comment (alongir) [reply helm-chart/values.yaml:15]: @kcns008 did you address @mertyildiran's change request?

**Review by mertyildiran — CHANGES_REQUESTED** [review]

File: helm-chart/templates/14-kubeshark-scc.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - persistentVolumeClaim
+  - secret
+  - hostPath
+  - projected
+  - ephemeral
+users:
+  - system:serviceaccount:{{ .Release.Namespace }}:kubeshark-service-account
```
Comment (mertyildiran) [inline helm-chart/templates/14-kubeshark-scc.yaml:51]:
```suggestion
{{- if .Capabilities.APIVersions.Has "security.openshift.io/v1/SecurityContextConstraints" }}
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
  labels:
    {{- include "kubeshark.labels" . | nindent 4 }}
  annotations:
    "helm.sh/hook": pre-install
  {{- if .Values.tap.annotations }}
    {{- toYaml .Values.tap.annotations | nindent 4 }}
  {{- end }}
  name: kubeshark-scc
priority: 10
allowPrivilegedContainer: true
allowHostDirVolumePlugin: true
allowHostNetwork: true
allowHostPorts: true
allowHostPID: true
allowHostIPC: true
readOnlyRootFilesystem: false
requiredDropCapabilities:
  - MKNOD
allowedCapabilities:
  - NET_RAW
  - NET_ADMIN
  - SYS_ADMIN
  - SYS_PTRACE
  - DAC_OVERRIDE
  - SYS_RESOURCE
  - CHECKPOINT_RESTORE
  - SYS_MODULE
runAsUser:
  type: RunAsAny
fsGroup:
  type: MustRunAs
seLinuxContext:
  type: RunAsAny
supplementalGroups:
  type: RunAsAny
seccompProfiles:
- '*'
volumes:
  - configMap
  - downwardAPI
  - emptyDir
  - persistentVolumeClaim
  - secret
  - hostPath
  - projected
  - ephemeral
users:
  - system:serviceaccount:{{ .Release.Namespace }}:kubeshark-service-account
{{- end }}
```

**Review by mertyildiran — CHANGES_REQUESTED** [review]

**Review by mertyildiran — APPROVED** [review]

**Pull request merged**